PR Template: Add test related checkboxes (#24787)
Automattic/jetpack: .github/PULL_REQUEST_TEMPLATE.md

* Go to '..'
*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

> - [ ] Have you checked the E2E test CI results, and verified that your changes does not break them?

Reviewer: ```suggestion
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
```

> - [ ] Have you written new tests for your changes, if applicable?
> - [ ] Have you checked the E2E test CI results, and verified that your changes does not break them?

Reviewer 2: I might re-word it in the first person, as long as there are check boxes to tick.

```suggestion
- [ ] I have written new tests for my changes / tests are not applicable
- [ ] I have checked the E2E tests CI results, and my changes do not break them
```

Reviewer 3: I don't think I like the first-person version. The stated intention here is "nudge", while first person checkboxes come across as much more of a requirement.

Author: Yup, I agree with Brad here, I'd like to keep it as a "nudge". We definitely can iterate on it later as needed.
